server-spi-private/src/main/java/org/keycloak/cookie/CookieType.java

```diff
@@ -5,11 +5,11 @@
 public final class CookieType {
 
     public static final CookieType AUTH_DETACHED = CookieType.create("KC_STATE_CHECKER")
-            .scope(CookieScope.LEGACY)
+            .scope(CookieScope.INTERNAL)
```

Comment:
Need to make sure this doesn't need to be available after a redirect to Keycloak. I don't think so, but @mposolda can you confirm?

Reply:
@stianst This cookie is used by info pages and error pages. It is used for the case when authenticationSession doesn't exists. It wraps the data about current info/error page being shown, so the user can change the locale and things still work (the error/info page is correctly displayed in the new language after user changed locale).

AFAIK especially the error pages can be shown right after redirect from the client. For instance when OIDC client redirects to Keycloak to send initial OIDC authentication request with invalid `redirect_uri` parameter, Keycloak displays the page like `Invalid redirect_uri` . The cookie needs to be available right in this error page response, so user is able to change the locale and display the error in different language. Will this work with `SameSite=Strict` ?

Reply:
@mposolda in this case doesn't KC set the cookie on this requests, rather than rely on existing values of the cookie?